packages/mui-material-next/src/Button/Button.tsx

```diff
@@ -348,7 +350,7 @@ export const ButtonRoot = styled('button', {
       backgroundColor: hoveredContainerColor[ownerState.variant ?? 'text'],
       boxShadow: hoveredContainerElevation[ownerState.variant ?? 'text'],
     },
-    '&:active': {
+    [`&.${buttonClasses.active}`]: {
```

Comment:
cc @siriwatknp we may need this change in Joy UI too. For e.g. see https://github.com/mui/material-ui/pull/35410 for bugs that could arise without using it.

Reply:
Got it.